include/rocksdb/file_system.h

```diff
@@ -733,6 +747,19 @@ struct FSReadRequest {
   IOStatus status;
 };
 
+// A read IO response structure for use by asynchronously Read APIs.
+struct FSReadResponse {
```

Comment:
I am not sure I understand why you need both the IOHandle and the FSReadResponse.  Shouldn't the IOHandle contain everything to identify the request (offset, length, status) and the callback to invoke when the read is complete?